### PR TITLE
initrd_size_decrease: remove mlxsw_spectrum/mlxsw_core kernel modules

### PR DESCRIPTION
**Pull request by ader1990**
This commit is part of the effort to decrease the initrd size: Partially-Fixes: https://github.com/flatcar/Flatcar/issues/1381

mlxsw_spectrum/mlxsw_core are kernel modules that enable Mellanox Technologies Switch ASICs support.
Removing the two modules frees ~5MB in the /boot partition and ~4MB in the vmlinuz.
`/dev/vda1          129039      59962      69078  47% /boot` with the removal on AMD64.

See discussion: https://github.com/flatcar/scripts/pull/1731#discussion_r1514213546

### CODE REVIEW

**Review by jepio — APPROVED** [review]

**Comment by jepio** [comment]
@ader1990 can you add a changelog entry?

**Comment by jepio** [comment]
Can you follow the example of other changelog entries and add a link to the PR? That way it's cross referenced and people can discover the motivation for a change (here: "reducing the size of the initrd").

**Comment by jepio** [comment]
And then feel free to merge directly since Kai already gave his approval.

File: changelog/changes/2024-03-12-remove-mlnx-switch-asics-drivers.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- Removed Linux drivers for Mellanox Technologies Switch ASICs family and Spectrum/Spectrum-2/Spectrum-3/Spectrum-4 Ethernet Switch ASICs.
```
Comment (pothos) [inline changelog/changes/2024-03-12-remove-mlnx-switch-asics-drivers.md:1]:
```suggestion
- Removed Linux drivers for Mellanox Technologies Switch ASICs family and Spectrum/Spectrum-2/Spectrum-3/Spectrum-4 Ethernet Switch ASICs to reduce the initrd size ([scripts#1734](https://github.com/flatcar/scripts/pull/1734))
```

Comment (ader1990) [reply changelog/changes/2024-03-12-remove-mlnx-switch-asics-drivers.md]: @pothos I added also the size freed and the issue number of the overall effort.

**Review by pothos — APPROVED** [review]

File: changelog/changes/2024-03-12-remove-mlnx-switch-asics-drivers.md (hunk 1 above)
Comment (pothos) [reply changelog/changes/2024-03-12-remove-mlnx-switch-asics-drivers.md]: Ok, it's a bit much detail for users but why not^^

**Pull request merged**